### ISSUE TITLE
Groups - containment vs. aggregation

### ISSUE THREAD

**Issue by deissnerk** [xregistry/spec#192]
Currently, the metamodel of xRegistry relies on the hierarchy of resources that are _owned_ by a group. If the group gets deleted, all resources it contains get deleted too. When referencing a resource with an href, a URL, or an xid, the reference always contains the group as part of the path. 

For V1, we need to decide, if we want to keep this concept of groups containing resources. Advanced features that could come after V1, may rely on uniquely identifying resources by the group they belong to. The schema of an xid will rely on it, as well as the REST API.

The containment might be what we want, as it allows to bundle together a group of resources and export, import, or delete it. On the other hand, an alternative would be a grouping concept that would be closer to tagging. Resources would be first-class citizens and could belong to one or multiple groups. 

This issue is created as a reminder that we need to take a conscious decision prior to V1. 
@duglin @Fannon

**Comment by duglin** [comment]
If we use tags, and then someone asked to see the entire registry as a single JSON doc, do you see a flat list of resources being returned or do you see it appearing more like it does today (as a hierarchy) where the same resource might appear in more than one branch of the tree if it is tagged with more than one group? I'm guessing it would be a flat list since tags are (almost) secondary bits of metadata at that point - and (I'm guessing) even optional.

**Comment by deissnerk** [comment]
@duglin Yes, I guess it would be a flat list. I'm not sure what I prefer. I only brought it up, because it came up in one of our calls. As I said above, we should think about it carefully, because it also has significant impact on the xid discussion. Do we think of resources as self-contained entities that could easily be moved around, or is the group part of its identity? If a resource is identified by its path (including group), moving it to another group would change its id and technically make it a different resource.

**Comment by Fannon** [comment]
The containment aspect of groups is fine, I think if we **understand groups to reflect "resource ownership" in the sense of who owns the responsibility to describe them**. Then it also makes more sense to remove or replace an entire group, because everything in it may be published and updated together anyway.

Where I see a problem is if we use the groups to express a taxonomy or the ownership of the models. Especially with taxonomy, a simple containment is just to inflexible and you'll quickly start to try to put too many concerns into one concept. Unfortunately, we are currently using the groups also as part of the model (the endpoint registry defines the group, not just the resource type). That means if we completely remove a group, it's more like an administrative action to remove an entire feature / content type from the registry.

**Comment by duglin** [comment]
@deissnerk close?

**Comment by duglin** [comment]
Closed since we now have "relation"
